tests/specs/js-errors/error-payload.e2e.js

```diff
@@ -0,0 +1,55 @@
+const config = {
+  init: {
+    privacy: { cookies_enabled: true }
+  }
+}
+
+describe('Error payloads', () => {
+  it('should add session replay flag if active', async () => {
+    await browser.url(await browser.testHandle.assetURL('instrumented.html', config)) // Setup expects before loading the page
+      .then(() => browser.waitForAgentLoad())
+
+    await browser.execute(function () {
+      Object.values(newrelic.initializedAgents)[0].runtime.session.state.sessionReplay = 1
+      newrelic.noticeError(new Error('test'))
+    })
+
+    const { request: { body: { err } } } = await browser.testHandle.expectErrors()
+
+    expect(err[0].params).toEqual(expect.objectContaining({
+      firstSeenAt: expect.any(Number),
+      hasReplay: true
+    }))
+  })
+
+  it('should NOT add session replay flag if not active', async () => {
+    await browser.url(await browser.testHandle.assetURL('instrumented.html')) // Setup expects before loading the page
+      .then(() => browser.waitForAgentLoad())
+
+    await browser.execute(function () {
+      newrelic.noticeError(new Error('test'))
+    })
+
+    const { request: { body: { err } } } = await browser.testHandle.expectErrors()
+
+    expect(Object.keys(err[0].params).includes('hasReplay')).toEqual(false)
+  })
+
+  it('should set a timestamp, tied to the FIRST error seen', async () => {
+    await browser.url(await browser.testHandle.assetURL('instrumented.html')) // Setup expects before loading the page
+      .then(() => browser.waitForAgentLoad())
+
+    const [before, after] = await browser.execute(function () {
+      var a = 0
+      var before = Date.now()
+      while (a++ < 20) newrelic.noticeError(new Error('test'))
```

Comment:
I'm not sure this test makes a lot of sense or is purposeful. The first error will have this param, but the rest 19 falls into sending `params.browser_stack_hash` instead.

I think really you should check that throwing an error (first) has this property with a reasonably correct timestamp (is not past the `after`), and then throwing the same error again _doesn't_ carry this property.

Also, note `noticeError` records the timestamp in a separate `handle('err',...)` than jserror module's `fn-err` cb and wrapped global handler -- both more likely to be what's hit if users run into unexpected errors: so it's better to test with `throw new Error(...)`.

Reply:
The way the error module works is that it aggregates many errors into a single object with a count.  This is just checking that of the 20 that threw, the timestamp should match the 1st only.

As for checking other errors containing the property, if thats a concern we should just test them all

Reply:
Thanks i think i see what you were meaning about the browser_stack_hash.  Will put in a more eloquent solution